test/lib/jdk/test/lib/hprof/parser/Reader.java

```diff
@@ -93,7 +93,6 @@ public static Snapshot readFile(String heapFile, boolean callStack,
                 return r.read();
             } else if ((i >>> 8) == GZIP_HEADER_MAGIC) {
                 // Possible gziped file, try decompress it and get the stack trace.
-                in.close();
```

Comment:
It is not obvious to me that there may not be a reason for closing all of the streams before opening new ones below.